src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -2812,7 +2818,7 @@ public void takeSnapshot(String tag, String... keyspaceNames) throws IOException
      */
     private void takeSnapshot(String tag, boolean ephemeral, String... keyspaceNames) throws IOException {
         if (operationMode == Mode.JOINING)
-            throw new IOException("Cannot snapshot until bootstrap completes");
+            logger.info("Taking snapshot of joining node");
```

Comment:
Make these messages a warn, with a notice that these snapshots are not complete / valid for a live cluster.